stripe/src/main/java/com/stripe/android/util/StripeTextUtils.java

```diff
@@ -81,6 +82,107 @@ public static boolean isBlank(String value) {
     }
 
     /**
+     * Returns a {@link CardBrand} corresponding to a partial card number, or {@link Card#UNKNOWN}
+     * if the card brand can't be determined from the input value.
+     *
+     * @param cardNumber a credit card number or partial card number
+     * @return the {@link CardBrand} corresponding to that number, or {@link Card#UNKNOWN} if it
+     * can't be determined
+     */
+    @NonNull
+    @CardBrand
+    public static String getPossibleCardType(@Nullable String cardNumber) {
+        if (isBlank(cardNumber)) {
+            return Card.UNKNOWN;
+        }
+
+        String spacelessCardNumber = convertToSpacelessNumber(cardNumber);
+
+        if (hasAnyPrefix(spacelessCardNumber, Card.PREFIXES_AMERICAN_EXPRESS)) {
+            return Card.AMERICAN_EXPRESS;
+        } else if (hasAnyPrefix(spacelessCardNumber, Card.PREFIXES_DISCOVER)) {
+            return Card.DISCOVER;
+        } else if (hasAnyPrefix(spacelessCardNumber, Card.PREFIXES_JCB)) {
+            return Card.JCB;
+        } else if (hasAnyPrefix(spacelessCardNumber, Card.PREFIXES_DINERS_CLUB)) {
+            return Card.DINERS_CLUB;
+        } else if (hasAnyPrefix(spacelessCardNumber, Card.PREFIXES_VISA)) {
+            return Card.VISA;
+        } else if (hasAnyPrefix(spacelessCardNumber, Card.PREFIXES_MASTERCARD)) {
+            return Card.MASTERCARD;
+        } else {
+            return Card.UNKNOWN;
+        }
+    }
+
+    /**
+     * Converts a card number that may have spaces between the numbers into one without any spaces.
```

Comment:
Can you add something to these like "doesn't verify that all digits match `\d`" to all of these `blahNumber(..)` methods?